internal/services/maintenance.go

```diff
@@ -0,0 +1,147 @@
+package services
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/jackc/pgerrcode"
+	"github.com/lib/pq"
+	"github.com/shulganew/gophermart/internal/config"
+	"github.com/shulganew/gophermart/internal/model"
+	"go.uber.org/zap"
+	"golang.org/x/crypto/bcrypt"
+)
+
+// User creation, registration, validation and autentification service
+type Maintenance struct {
+	stor Maintenancerer
+}
+
+type Maintenancerer interface {
+	AddUser(ctx context.Context, login string, hash string) (*uuid.UUID, error)
+	GetByLogin(ctx context.Context, login string) (*model.User, error)
+}
+
+func NewRegister(stor Maintenancerer) *Maintenance {
```

Comment:
тут не только регистрация. Давай назовем файл user.go и это будет userService
```suggestion
func NewUserService(stor Maintenancerer) *UserService {
```

Reply:
Готово